docs/serving/using-knative-services.md

```diff
@@ -0,0 +1,233 @@
+---
+title: "User-created Knative Services"
+weight: 01
+linkTitle: "User-created Knative Services"
+type: "docs"
+---
+
+Knative services are Kubernetes Services that a user creates to deploy an application. Each Knative Service is defined by a Route and a Configuration, which have the same name as the Service, contained in a YAML file.
+
+## Before you begin
+
+To use Knative Services, you will need:
+* A Kubernetes cluster with [Knative installed](https://knative.dev/docs/install/index.html).
+
+## Creating a Knative Service
+To create an application, you need to create a YAML file that defines a Knative Service.
+This YAML file specifies metadata about the application, points to the hosted image of the app for deployment, and allows the deployment to be configured.
+
+This guide uses the [Hello World sample app in Go](https://knative.dev/docs/serving/samples/hello-world/helloworld-go) to demonstrate the structure of a Service YAML file and the basic workflow for deploying an app, but these steps can be adapted for your own application if you have an image of it available on Docker Hub, Google Container Registry, or another container image registry.
+
+The Hello World sample app does the following:
+1. reads in an env variable, TARGET, from the configuration .yaml file
+2. prints “Hello World: \${TARGET}!“. If TARGET isn’t defined, it will print “NOT SPECIFIED”.
+
+### Procedure
+
+1. Create a new file named `service.yaml` containing the following information.
+
+```yml
+apiVersion: serving.knative.dev/v1 (1)
+kind: Service
+metadata:
+  name: helloworld-go (2)
+  namespace: default (3)
+spec:
+  template:
+    spec:
+      containers:
+        - image: gcr.io/knative-samples/helloworld-go (4)
+          env:
+            - name: TARGET (5)
+              value: "Go Sample v1"
+```
+**(1)** Current version of Knative   
+**(2)** The name of the app   
+**(3)** The namespace the app will use   
+**(4)** The URL to the image of the app   
+**(5)** The environment variable printed out by the sample app
+
+**NOTE:** If you’re deploying an image of your own app, update the name of the app and the URL of the image accordingly.
+
+## Creating a private cluster-local Service
+By default services deployed through Knative are published to an external IP
+address, making them public services on a public IP address and with a
+[public URL](./using-a-custom-domain.md).
```

Comment:
Probably worth noting here that the public URL out of the box is "example.com" which is non-functioning as it does not point to the public IP address.

Reply:
I think this is covered in / under the scope of https://knative.dev/docs/serving/using-a-custom-domain/ maybe?